### PR TITLE
allow admin to disable lookupserver upload

### PR DESCRIPTION
**Pull request by schiessle**
Allow the admin to disable the possibility to publish user specific data on the lookup server.

fixes the "turn off showing phone number, address, WWW, Twitter" part of https://github.com/nextcloud/server/issues/3821 

cc @KB7777

### CODE REVIEW

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

File: apps/federatedfilesharing/lib/FederatedShareProvider.php [hunk 1]
```diff
@@ -963,4 +963,15 @@ public function isLookupServerQueriesEnabled() {
...
+
+
+	/**
+	 * Check if it is allowed to publish user specific data to the lookup server
+	 *
+	 * @return bool
+	 */
+	public function isLookupServerUploadEnabled() {
+		$result = $this->config->getAppValue('files_sharing', 'lookupServerUploadEnabled', 'yes');
+		return ($result === 'yes') ? true : false;
```
Comment (ChristophWurst) [inline apps/federatedfilesharing/lib/FederatedShareProvider.php]: the `($result === 'yes')` part already results in a proper boolean value, no need for the ternary operator here 😉 

We could even shorten this to
```php
return $this->config->getAppValue('files_sharing', 'lookupServerUploadEnabled', 'yes') === 'yes';
```

**Review by ChristophWurst — CHANGES_REQUESTED** [review]
some nitpicks, the rest looks good though

File: apps/federatedfilesharing/lib/FederatedShareProvider.php (hunk 1 above)
Comment (schiessle) [inline apps/federatedfilesharing/lib/FederatedShareProvider.php]: completely right, that's happen if you copy&paste methods and only replace the relevant parts 😉 

I will not do the last step you suggested because I think this makes it harder to debug but will simplify the return statement

**Review by ChristophWurst — APPROVED** [review]
